### PR TITLE
Add IGetAll<> interface to the IChannelRepository

### PR DESCRIPTION
**Pull request by zentron**
We support the `/all` endpoint in[ OctopusServer for Channels](https://github.com/OctopusDeploy/OctopusDeploy/blob/1f1a7a667daa86a14e8dee4f448d346c3b673389/source/Octopus.Server/Web/Api/Actions/Channels/ChannelsNancyModule.cs#L27) so in to synchronize the surface area we should probably ensure the C# client can interact with it.

### CODE REVIEW

File: build/Build.cs [hunk 1]
```diff
@@ -170,14 +181,31 @@ private void SignBinaries(AbsolutePath path)
     {
         Info($"Signing binaries in {path}");
         var files = path.GlobDirectories("**").SelectMany(x => x.GlobFiles("Octopus.*.dll"));
+
```
Comment (zentron) [inline build/Build.cs:184]: These unrelated changes were required to have the build complete successfully.

**Review by uglybugger — APPROVED** [review]
LGTM :)
